### PR TITLE
nimpretty: fix #11699, correct indentation when splitting containers

### LINKED ISSUE #11699: Nimpretty issue in lists over 80 chars.
This list is long:

```
const keywords = @[
  "foo", "bar", "foo", "bar", "foo", "bar", "foo", "bar", "foo", "bar", "foo", "bar", "foo", "bar",
  "zzz", "ggg", "ddd",
]
```

`nimpretty bug.nim` introduces a strange indentation space where there should not be.

```
const keywords = @[
  "foo", "bar", "foo", "bar", "foo", "bar", "foo", "bar", "foo", "bar", "foo",
      "bar", "foo", "bar",
  "zzz", "ggg", "ddd",
]
```

I expected some thing like:

```
const keywords = @[
  "foo", "bar", "foo", "bar", "foo", "bar", "foo", "bar", "foo", "bar", "foo",
  "bar", "foo", "bar",
  "zzz", "ggg", "ddd",
]
```

### PR DESCRIPTION
**Pull request by narimiran**

### CODE REVIEW

File: compiler/layouter.nim [hunk 1]
```diff
@@ -44,6 +44,7 @@ type
     col, lastLineNumber, lineSpan, indentLevel, indWidth*, inSection: int
     keepIndents*: int
     doIndentMore*: int
+    parIndent: bool
```
Comment (Araq) [inline compiler/layouter.nim]: wouldn't it be better to count this? Else ')' closes ALL parIndents.

File: nimpretty/tests/expected/exhaustive.nim [hunk 2]
```diff
@@ -686,8 +686,7 @@ block:
 proc newRecordGen(ctx: Context; typ: TypRef): PNode =
   result = nkTypeDef.t(
     newId(typ.optSym.name, true, pragmas = [id(
-        if typ.isUnion: "cUnion" else: "cStruct")]),
-    empty(),
+    if typ.isUnion: "cUnion" else: "cStruct")]), empty(),
```
Comment (Araq) [inline nimpretty/tests/expected/exhaustive.nim]: but this is worse

Comment (narimiran) [reply nimpretty/tests/expected/exhaustive.nim]: Yes, it is worse, but it wasn't human-pretty even before. I took it as positive trade-off: slightly worse here, but it makes other code much nicer.

**Comment by narimiran** [comment]
It was solved differently in https://github.com/nim-lang/Nim/commit/7deb49e992b734d5ae513460fdc9cf68d4000588

**Pull request closed without merging**